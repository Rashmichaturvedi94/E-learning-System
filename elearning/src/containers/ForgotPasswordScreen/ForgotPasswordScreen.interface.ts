export interface ForgotPasswordScreenProps {
  // TODO: add required props below
  myProp?: boolean;
}
