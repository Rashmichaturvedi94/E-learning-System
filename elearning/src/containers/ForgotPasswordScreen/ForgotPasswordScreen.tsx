import React, { useState } from 'react';
import { Alert, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import auth from '@react-native-firebase/auth';
import {
  Container,
  Input,
  TitleText,
  ContinueButton,
  SaveText,
  InfoText,
} from './ForgotPasswordScreen.styles';

export const ForgotPasswordScreen = () => {
  const [email, setEmail] = useState('');
  const handleResetPassword = () => {
    auth()
      .sendPasswordResetEmail(email)
      .then(() => {
        Alert.alert('Email sent', '');
      })
      .catch((error) => {
        Alert.alert(error.message.split(']').pop());
      });
  };

  return (
    <Container>
      <View>
        <TitleText>Forgot Password?</TitleText>
        <InfoText>
          Confirm your email and we will send you instructions.
        </InfoText>
        <Input
          placeholder="Email"
          onChangeText={(text) => setEmail(text)}
          value={email.toLowerCase()}
          autoCapitalize="none"
          keyboardType="email-address"
          autoCompleteType="email"
        />
        <ContinueButton onPress={handleResetPassword}>
          <SaveText>Submit</SaveText>
        </ContinueButton>
      </View>
    </Container>
  );
};
