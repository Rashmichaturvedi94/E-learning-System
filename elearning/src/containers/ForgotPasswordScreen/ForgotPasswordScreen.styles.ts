import styled from 'styled-components/native';
import { StyleSheet, SafeAreaView, TextInput, Text, TouchableHighlight } from 'react-native';
import { ForgotPasswordScreenProps } from './ForgotPasswordScreen.interface';


export const Container = styled(SafeAreaView)`
  flex: 1;
  justify-content: center;
  margin: 0px 16px;
`;

export const TitleText = styled(Text)`
  text-align: center;
  font-size: 30px;
  font-weight: bold;
  margin-bottom: 50px;
`;


export const InfoText = styled(Text)`
  font-size: 18px;
  margin-top: 10px;
  text-align: center;
  color: #9a9a9c;
`;

export const Input = styled(TextInput)`
  height: 40px;
  margin-top: 20px;
  border-bottom-color: #b0adac;
  border-bottom-width: 0.5px;
  border-color: white;
`;

export const ContinueButton = styled(TouchableHighlight)`
  height: 50px;
  border-radius: 10px;
  background-color: #1b8ffa;
  margin-top: 100px;
  justify-content: center;
  align-items: center;
`;

export const SaveText = styled(Text)`
  color: #FFFFFF;
`;
