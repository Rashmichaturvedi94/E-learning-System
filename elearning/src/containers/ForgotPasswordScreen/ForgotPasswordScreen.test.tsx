import React from 'react';
import { render } from '@testing-library/react-native';
import { ForgotPasswordScreen } from './ForgotPasswordScreen';

describe('containers/ForgotPasswordScreen', () => {
  it('should render', () => {
    const component = render(<ForgotPasswordScreen />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
