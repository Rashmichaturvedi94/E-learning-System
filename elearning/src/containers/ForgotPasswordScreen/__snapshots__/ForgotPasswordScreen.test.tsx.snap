// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/ForgotPasswordScreen should render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 64,
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 50,
            "textAlign": "center",
          },
        ]
      }
    >
      Forgot Password?
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9a9a9c",
            "fontSize": 18,
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      Confirm your email and we will send you instructions.
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCompleteType="email"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomColor": "#b0adac",
            "borderBottomWidth": 0.5,
            "borderColor": "white",
            "height": 40,
            "marginTop": 20,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1b8ffa",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 100,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
